crates/ruff_cli/src/diagnostics.rs

```diff
@@ -203,6 +207,18 @@ pub(crate) fn lint_path(
 
     let imports = imports.unwrap_or_default();
 
+    if let Some((package_cache, relative_path, file_last_modified)) = caching {
+        if parse_error.is_some() {
+            // Currently we don't cache parsing error, so we remove the old
+            // file cache (if any).
+            // TODO: cache parse errors?
```

Comment:
i don't think we need to cache parse errors

Reply:
:+1: will remove the TODO.